frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -360,6 +360,7 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 			if d.stage.BaseName == emptyImageName {
 				d.state = llb.Scratch()
 				d.image = emptyImage(platformOpt.targetPlatform)
+				d.platform = &platformOpt.targetPlatform
```

Comment:
Looks like this can be nil pointer dereference. In that case we should use `DefaultSpec`

Reply:
Ah, I read it wrong. This is not even a pointer, so it can't be nil.